mlrun/feature_store/retrieval/job.py

```diff
@@ -93,18 +96,25 @@ def set_default_resources(resources, setter_function):
         set_default_resources(
             function.spec.executor_resources, function.with_executor_requests
         )
+    if start_time and not isinstance(start_time, str):
+        start_time = start_time.isoformat()
+    if end_time and not isinstance(end_time, str):
+        end_time = end_time.isoformat()
```

Comment:
I guess this is ok, so long as we're aware that, if the type is wrong, this will raise a slightly confusing (but understandable by a dev) error that type X does not have a method called `isoformat`.